### PR TITLE
feat(kit): `addTypeTemplate` helper with auto-registration

### PR DESCRIPTION
**Pull request by Hebilicious**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR adds an `addTypes` utility to `@nuxt/kit`.
It also improves the documentation section regarding adding types, by including a general usecase for `addTemplate` and documented an use-case of `updateTemplate`.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/kit/src/template.ts [hunk 1]
```diff
@@ -13,16 +13,38 @@ export function addTemplate (_template: NuxtTemplate<any> | string) {
...
 
   // Add to templates array
   nuxt.options.build.templates.push(template)
 
+  // Add template to types reference
+  if (template.isType === true) {
+    nuxt.hook('prepare:types', ({ references }) => {
+      references.push({ path: template.filename })
+    })
+  }
```
Comment (danielroe) [inline packages/kit/src/template.ts:37]: Let's move this into `addTypes` and leave `addTemplate` untouched.

Comment (Hebilicious) [reply packages/kit/src/template.ts]: I refactored accordingly

**Review by danielroe — APPROVED** [review]

**Comment by Hebilicious** [comment]
@danielroe I forgot to click submit review 🤦🏽  I can open another PR for the typos

**Comment by danielroe** [comment]
Thank you 😊

**Pull request merged**